### PR TITLE
Release - Break release job apart

### PR DESCRIPTION
**Pull request by araratthehero**
## Description
Break release job apart into multiple jobs.
[Here](https://github.com/Adyen/adyen-android/actions/runs/11572437317) is a workflow run which demonstrated how it looks.

## Checklist 
- [x] PR is labelled 
- [x] Changes are tested manually

COAND-1013

### CODE REVIEW

File: .github/workflows/publish_to_maven_central.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 17
+
+      - name: Gradle cache
```
Comment (OscarSpruit) [inline .github/workflows/publish_to_maven_central.yml:19]: I know we had it before, but should we remove caching for releases?

Comment (araratthehero) [reply .github/workflows/publish_to_maven_central.yml:19]: I think the cache can help us execute gradle tasks faster. I did a small experiment with it.

I just ran different workflows with and without cache.
[Workflow with cache](https://github.com/Adyen/adyen-android/actions/runs/11591406960): It looks like it saves us 3.5 minutes on average.
[Workflow without cache](https://github.com/Adyen/adyen-android/actions/runs/11590456730): The cache has been removed from `check` and `publish_to_maven_central` jobs. The first thing I see is that it downloads the gradle, which IMO takes the extra time. 
image

Comment (OscarSpruit) [reply .github/workflows/publish_to_maven_central.yml:19]: Yes I understand it takes more time, but if the cache is ever corrupt we might publish a build that is broken

Comment (araratthehero) [reply .github/workflows/publish_to_maven_central.yml:19]: Hmm, I see your point. I will remove it then so we are safe.

**Review by OscarSpruit — APPROVED** [review]

**Pull request merged**